dlr-gtlab/gtlab-core
# Gtobjectui regexp validation using objects

Currently the validation for renaming of ObjectUI elements have no access to the underlying object. This is added here with new interface functions.

A new kind of regular expression is added to be not so strict with intermediate states of the renaming. It is allowed to type everything and only in the end this is checked.
As this allows more freedom it is also hared to detect problems while renaming.
Therefore it is up the the programmer to use this new system or a version which is closer to the old behavior by a flag.

And additionally the objectUI can also define a hint how the regular expression limits the renaming as regular expressions are not readable for users that good.

## How Has This Been Tested?
It has been tested by a implementation in the performance module to use this new features and in some basic UI elements to check if the behavior keeps the same for the other classes.

## Hint for reviwers how to test usage:
The test modules of GTlab should be build and a project using the testing modules ("TestDatamodelInterface") should be opened.
- Select an object of the class "TestDmiExternalObject"
- Open the context menu there and select "Rename"
- The renaming is restricted to the reg exp for letters, numbers few brackets and spaces. So if you use a sign like a "." the text should become orange and should not be able to be used for the renaming. 
- For existing implementations nothing should be changed while renaming but the hint in the logging widget with the hint which restrictions are applied if an invalid sign is used.

## Checklist:


- [x] A test for the new functionality was added (if applicable).
- [x] All tests run without failure.
- [ ] The changelog has been extended, if this MR contains important changes from the users's point of view.
- [x] The new code complies with the GTlab's style guide.
- [ ] New interface methods / functions are exported via `EXPORT`. Non-interface functions are NOT exported.
- [x] The number of code quality warnings is not increasing.
